pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -42,8 +42,14 @@ let
   # The wrapper scripts use 'cat' and 'grep', so we may need coreutils.
   coreutils_bin = if nativeTools then "" else getBin coreutils;
 
-  default_cxx_stdlib_compile = optionalString (targetPlatform.isLinux && !(cc.isGNU or false) && !nativeTools)
-    "-isystem $(echo -n ${cc.gcc}/include/c++/*) -isystem $(echo -n ${cc.gcc}/include/c++/*)/$(${cc.gcc}/bin/gcc -dumpmachine)";
+  default_cxx_stdlib_compile = if (targetPlatform.isLinux && !(cc.isGNU or false) && !nativeTools) then
+      "-isystem $(echo -n ${cc.gcc}/include/c++/*) -isystem $(echo -n ${cc.gcc}/include/c++/*)/$(${cc.gcc}/bin/gcc -dumpmachine)"
+    else if (targetPlatform.isDarwin && (cc.isClang or true) && !nativeTools && libcxx != null) then
+      "-isystem ${libcxx}/include/c++/v1"
+    else
+      "";
+  default_cxx_stdlib_link = optionalString (targetPlatform.isDarwin && (cc.isClang or true) && !nativeTools && libcxx != null)
+    "-L${libcxx}/lib -stdlib=libc++";
```

Comment:
I think there's a function to get the library directory of a derivation, I forget what it was though (`getLib` maybe), and whether it should be used in this situation.

Reply:
Yeah it's probably safer to use getLib here. libc++ could become multiple outputs in the future.